### ISSUE TITLE
SVG: fix marker bounding box logic

### ISSUE THREAD

**Issue by artemp** [mapnik/mapnik#3224]
The bellow SVG fails to render with `svg2png` because reported bounding box has 0 height.
e.g `found width of '220' and height of '0'``
`Exception caught:Could not write to empty stream`

```svg
<svg width="300px" height="200px" viewBox="0 0 300 200">

    <line x1="40" x2="260" y1="100" y2="100" stroke="#5184AF" stroke-width="20" stroke-linecap="round" stroke-dasharray="1, 30"/>

</svg>
```

**Comment by artemp** [comment]
@jakepruitt - could you take a look and see why `height/width` and `viewBox="0 0 300 200"` is not used in marker bounding box logic ?

**Comment by springmeyer** [comment]
See also #1122

**Comment by artemp** [comment]
Should we change to ? /cc @springmeyer @jakepruitt

diff --git a/include/mapnik/marker.hpp b/include/mapnik/marker.hpp
index 3aaa1cd..d8e1b12 100644
--- a/include/mapnik/marker.hpp
+++ b/include/mapnik/marker.hpp
@@ -115,11 +115,11 @@ public:

     inline double width() const
     {
-        return vector_data_->bounding_box().width();
+        return vector_data_->width();
     }
     inline double height() const
     {
-        return vector_data_->bounding_box().height();
+        return vector_data_->height();
     }

     mapnik::svg_path_ptr get_data() const

**Comment by jakepruitt** [comment]
@artemp yes, I think that would fix it. I am not sure how `bounding_box` is calculated, but I know `width` and `height` are based on the values in the `svg` element.

**Comment by springmeyer** [comment]
/cc @bsudekum likely just hit this. The basic issue is that we use the dynamically calculated values of geometries to determine height and width because this has seemed the most predictable. But this can fail depend on usage. For example @bsudekum's usage was having a bunch of separate graphics in one inkscape file and then selecting each and exporting. That type of export from inkscape seemed to keep all the svg paths in the file still which meant that mapnik would get the bounds for all of them... (I think is what was happening).
